docs/quickstart/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,926 @@
+Let's create an [abi](https://docs.soliditylang.org/en/v0.8.13/abi-spec.html#:~:text=Contract%20ABI%20Specification-,Basic%20Design,as%20described%20in%20this%20specification.) of our solidity code.
+
+In the same `./contract-examples/contracts` directory, let's run
+
+```
+solc --abi IHelloWorld.sol -o .
+```
+
+This spits out the abi code!
+
+IHelloWorld.abi
+
+```json
+[
+  {
+    "inputs": [],
+    "name": "sayHello",
+    "outputs": [
+      { "internalType": "string", "name": "result", "type": "string" }
+    ],
+    "stateMutability": "nonpayable",
+    "type": "function"
+  },
+  {
+    "inputs": [
+      { "internalType": "string", "name": "recipient", "type": "string" }
+    ],
+    "name": "setGreeting",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
+  }
+]
+```
+
+Note: The abi must have named outputs in order to generate the precompile template.
+
+### Generating the Precompile
+
+Now that we have an abi for the precompile gen tool to interact with. We can run the following command to generate our HelloWorld precompile!
+
+In the root of the repo run
+
+```bash
+go run ./cmd/precompilegen/main.go --abi ./contract-examples/contracts/IHelloWorld.abi --type HelloWorld --pkg precompile --out ./precompile/hello_world.go
+```
+
+Wow! We just got a precompile template that's mostly complete located at `./precompile/hello_world.go`. Let's fill out the rest!
+
+The precompile gives us commented instructions on the first 25 lines of the autogenerated file. Let's look at the 10 steps and follow them step by step.
+
+```go
+/* General guidelines for precompile development:
+1- Read the comment and set a suitable contract address in precompile/params.go. E.g:
+	HelloWorldAddress = common.HexToAddress("ASUITABLEHEXADDRESS")
+2- Set gas costs here
+3- It is recommended to only modify code in the highlighted areas marked with "CUSTOM CODE STARTS HERE". Modifying code outside of these areas should be done with caution and with a deep understanding of how these changes may impact the EVM.
+Typically, custom codes are required in only those areas.
+4- Add your upgradable config in params/precompile_config.go
+5- Add your precompile upgrade in params/config.go
+6- Add your solidity interface and test contract to contract-examples/contracts
+7- Write solidity tests for your precompile in contract-examples/test
+8- Create your genesis with your precompile enabled in tests/e2e/genesis/
+9- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+*/
+```
+
+### Step 1: Set Contract Address
+
+In `./precompile/hello_world.go`, we can see our precompile address is set to some default value. We can cut the address from the var declaration block and remove it from the precompile.
+
+![Singleton StatefulPrecompiledContract and signatures](./../../static/img/2022-09-01-22-46-00.png)
+
+We can paste it here in `./precompile/params.go`.
```

Comment:
Please be more specific on the exact location where to paste. There are multiple `ADD YOUR PRECOMPILE HERE` in the file. 

Please also provide an end result after the cut+paste.

I am already lost by here. I will wait for your changes before reviewing more.